mrtt-ui/src/library/mapDataForApi.js

```diff
@@ -0,0 +1,42 @@
+import { questionMapping } from '../data/questionMapping'
+import separateOutliersFromInliers from './separateOutliersFromInliers'
+
+const getFormattedAnswer = (questionMetaData, uiAnswer) => {
+  const { questionType, options } = questionMetaData
+
+  if (questionType === 'singleChoice') {
+    return [{ choice: uiAnswer }]
+  }
+  if (questionType === 'multipleChoiceOther') {
+    const { inliers, outliers } = separateOutliersFromInliers({
+      inliersList: options,
+      listContainingOutliers: uiAnswer
+    })
+    const isValidOtherValue = !!outliers[0]
+    const otherValueForApi = { choice: 'Other', value: outliers[0] }
+    const nonOtherValuesForApi = inliers.map((nonOtherValue) => ({ choice: nonOtherValue }))
+
+    return isValidOtherValue ? [...nonOtherValuesForApi, otherValueForApi] : nonOtherValuesForApi
+  }
+
+  return uiAnswer
+}
+// set up data structure for api using question_id and answer_value object
+// ensure formTitle is the same is title in questionMapping
+export const mapDataForApi = (formTitle, data) => {
+  const preppedData = []
+
+  for (const [fieldName, uiAnswer] of Object.entries(data)) {
+    const sectionQuestionsMetaData = questionMapping[formTitle]
+    const questionMetaData = sectionQuestionsMetaData?.[fieldName]
+    if (questionMetaData) {
+      preppedData.push({
+        question_id: questionMetaData.apiQuestionId,
+        answer_value: getFormattedAnswer(questionMetaData, uiAnswer)
```

Comment:
I needed to pass the 'questionMetaData` object to format multiple choice other questions anyway, so refactored to test for that, instead of hasOwnProperties stuff. Did some var renaming to wrap my head around it. What do you think?

Reply:
can we rename `questionMetaData` to `questionNumber `? or `questionId`?

Reply:
The id at the very least